### PR TITLE
Perch enc

### PR DESCRIPTION
**Pull request by yaakovpraisler**
## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/20295

## Description
Adding soc_status parameter to filter fetched incidents by their status

## Minimum version of Demisto
- [x] 5.0.0
- [ ] 5.5.0
- [ ] 6.0.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [x] No

## Must have
- [x] Tests
- [ ] Documentation

### CODE REVIEW

File: Packs/Perch/Integrations/Perch/Perch.py [hunk 1]
```diff
@@ -433,8 +441,15 @@ def create_indicator_command():
 def fetch_alerts(last_run, headers):
     last_fetch = last_run.get('time')
     url = '/alerts'
-    res = http_request('GET', url, headers=headers)
-    items = res.get('results')
+    statuses_to_fetch = demisto.params().get('soc_status', [])
+    if statuses_to_fetch:
+        items = []
+        for status in statuses_to_fetch:
```
Comment (roysagi) [inline Packs/Perch/Integrations/Perch/Perch.py:447]: is there no way to fetch several statuses with a single call?

Also if there is a get-alerts style command we should make sure it supports status filtering as well.

File: Packs/Perch/ReleaseNotes/1_0_1.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+
+#### Integrations
+##### Perch
+- Added filter to fetch incidents which enables you to filter fetched incidents by their soc_status.
```
Comment (roysagi) [inline Packs/Perch/ReleaseNotes/1_0_1.md:4]:
```suggestion
- Added support for the *Incident Soc Statuses to Fetch*  integration parameter that enables filtering fetched incidents by their soc_status.
```

**Review by roysagi — CHANGES_REQUESTED** [review]

**Comment by roysagi** [comment]
@yaakovpraisler - Also please add a unit test.

File: Packs/Perch/Integrations/Perch/Perch.py (hunk 1 above)
Comment (yaakovpraisler) [reply Packs/Perch/Integrations/Perch/Perch.py:447]: No, the API gets the parameter only as a string.

There is a command *perch-search-alerts* and it's supports the soc_status filtering.

**Review by roysagi — COMMENTED** [review]
@yaakovpraisler - Please add a unit test for the change in fetch.

**Review by roysagi — APPROVED** [review]
Good Job!

**Comment by roysagi** [comment]
@yaakovpraisler - please just add the new integration param to the README.md as well.

**Pull request merged**